### PR TITLE
Help detect typos in device add phrase

### PR DESCRIPTION
**Pull request by patrickxb**
Checks words entered are in list
Checks number of words
Short timeout on Hello, custom error message

Fixes CORE-2232

r? @maxtaco

### CODE REVIEW

File: go/kex2/provisioner.go [hunk 1]
```diff
@@ -149,6 +154,9 @@ func (p *provisioner) runProtocolWithCancel() (err error) {
 		p.canceled = true
 		return ErrCanceled
 	case err = <-ch:
+		if _, ok := err.(rpc.CanceledError); ok && !p.helloReceived {
+			return ErrHelloTimeout
```
Comment (maxtaco) [inline go/kex2/provisioner.go]: does this error export properly over RPC? We'll likely need it for the desktop client, I'm guessing.

Comment (patrickxb) [inline go/kex2/provisioner.go]: That's not the error that goes over RPC.  The engine checks for it and returns its own error.  But the engine error isn't exportable, so I'll fix that.

File: go/libkb/secwords.go [hunk 2]
```diff
@@ -30,6 +35,28 @@ func secWordListN(n int) ([]string, error) {
 	return res, nil
 }
 
+func validPhrase(p string, entropy int) error {
+	numWords := secWordCount(entropy)
```
Comment (maxtaco) [inline go/libkb/secwords.go]: what happens if you mistakenly type 2 spaces between a word?

File: go/engine/device_add.go [hunk 3]
```diff
@@ -116,6 +118,9 @@ func (e *DeviceAdd) Run(ctx *Context) (err error) {
 
 	// run provisioner
 	if err = RunEngine(provisioner, ctx); err != nil {
+		if err == kex2.ErrHelloTimeout {
+			return errors.New("Failed to provision device: are you sure you typed the secret properly?")
```
Comment (patrickxb) [inline go/engine/device_add.go]: Change to exportable error.

File: go/libkb/secwords.go (hunk 2 above)
Comment (patrickxb) [inline go/libkb/secwords.go]: that's a typo and you get an error

Comment (maxtaco) [inline go/libkb/secwords.go]: ok as long as the checker agrees with the stretcher!

**Comment by maxtaco** [comment]
Looks great. :+1: 

My only reservation is, did we pick the right timeout? Though I see arguments for and against.

**Comment by patrickxb** [comment]
Not sure.  We had the one bug report where the user had a really slow connection to keybase.io...so maybe 15s isn't enough.

**Pull request merged**